vmdb/spec/models/ems_refresh/refreshers/foreman_refresher_spec.rb

```diff
@@ -12,18 +11,26 @@
                       )
   end
 
-  # these values are from the
-  let(:provisioning_manager) { Provider.first.provisioning_manager }
-  let(:osfs) { provisioning_manager.operating_system_flavors }
+  let(:provisioning_manager)  { provider.provisioning_manager }
+  let(:osfs)                  { provisioning_manager.operating_system_flavors }
   let(:customization_scripts) { provisioning_manager.customization_scripts }
-  let(:media) { provisioning_manager.customization_script_media }
-  let(:ptables) { provisioning_manager.customization_script_ptables }
-  let(:configuration_manager) { Provider.first.configuration_manager }
+  let(:media)                 { provisioning_manager.customization_script_media }
+  let(:ptables)               { provisioning_manager.customization_script_ptables }
+  let(:configuration_manager) { provider.configuration_manager }
 
   it "will perform a full refresh on api v2" do
+    # Stub the queueing of the refresh so that when the manager
+    #  queues up an alternate refresh we will execute it immediately.
     EmsRefresh.stub(:queue_refresh) { |*args| EmsRefresh.refresh(*args) }
+
     VCR.use_cassette("#{described_class.name.underscore}_api_v2") do
-      EmsRefresh.queue_refresh(provider)
+      EmsRefresh.refresh(provider)
+
+      configuration_manager.reload
+      expect(configuration_manager.last_refresh_error).to be_nil
```

Comment:
@blomquisg I want this on *all* the refresher specs.  This way if a failure happens, then instead of the "unused http interactions" VCR error, you get the *actual* error.  Plus, it's a good double check anyway.